### DISCUSSION TITLE
Marine ecoregions

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#577] [Q&A]
Ok. I'm working on a portfolio post for the species migration assignment. I chose humpback whales as a species to map. I can't use the same ecoregions shapefile that Elsa used for monarch butterflies in the video because whales are a marine species. I searched for a marine ecoregions shapefile and found one here: https://databasin.org/datasets/3b6b12e7bcca419990c9081c0af254a2/

The issue is, I don't understand how to input this into the code.  Here's the original code cell inputting the terrestrial ecoregions, which ran just fine:
![image](https://github.com/user-attachments/assets/973ebfc7-68ec-4e5d-bb1b-ae9ec3781438)

And this is how I modified it to try to input the marine ecoregions:
![image](https://github.com/user-attachments/assets/fdcf94a8-c1a3-4f02-8235-5113cb63a52e)

The above didn't work and gave me a "DataSourceError", saying that the URL is not recognized as a supported file format. It seems like the issue is because I'm linking to a URL and not to a zip... in the original version, how was Elsa able to link to a zip?

**Comment by allenmoench** [comment]
Looking at this again, it seems like maybe the reason this isn't working is that there isn't actually a shapefile here. If there were, I could presumably right click the link to the shapefile > copy link address > paste it in the notebook. 

I did find another website which does have a shapefile here https://marineregions.org/downloads.php, but they won't let you link to the shapefile. I was able to download it though. I feel like there should be a way to put the downloaded file into the code, but I'm not sure what it is. I'll keep trying to puzzle it out, but in the meantime if anyone can let me know how to do that it would be much appreciated!

**Reply by cmq879** [reply]
I found this website: https://www.worldwildlife.org/publications/marine-ecoregions-of-the-world-a-bioregionalization-of-coastal-and-shelf-areas
which has the following link for a MEOW shapefile: https://files.worldwildlife.org/wwfcmsprod/files/Publication/file/7gger8mmke_MEOW_FINAL.zip
You could try putting that into your code defining eco_url and seeing if it will download to the virtual machine.
